File: 프로그래머스/3/42898. 등굣길/등굣길.java
class Solution {
    private static final int PUDDLE = 1;
    private static final int MOD = 1_000_000_007;

    public int solution(int m, int n, int[][] puddles) {
        int i;
        int j;
        int len;
        int[][] dp;

        dp = new int[m + 1][n + 1];
        len = puddles.length;
        for (i = 0; i < len; i++) {
            dp[puddles[i][0]][puddles[i][1]] = PUDDLE;
        }
        dp[0][1] = 1;
        for (i = 1; i <= m; i++) {
            for (j = 1; j <= n; j++) {
                if (dp[i][j] == PUDDLE) {
                    dp[i][j] = 0;
                } else {
                    dp[i][j] = (dp[i - 1][j] + dp[i][j - 1]) % MOD;
                }
            }
        }
        return dp[m][n];
    }
}

File: 프로그래머스/3/42898. 등굣길/등굣길.cpp
#include <string>
#include <vector>

using namespace std;

constexpr int PUDDLE = 1;
constexpr int MOD = 1'000'000'007;

int solution(int m, int n, vector<vector<int>> puddles) {
    int i;
    int j;
    int size;
    vector<vector<int>> map;

    map.resize(m + 1, vector<int>(n + 1, 0));
    size = puddles.size();
    for (i = 0; i < size; i++) {
        map[puddles[i][0]][puddles[i][1]] = PUDDLE;
    }
    map[0][1] = 1;
    for (i = 1; i <= m; i++) {
        for (j = 1; j <= n; j++) {
            if (map[i][j] == PUDDLE) {
                map[i][j] = 0;
            } else {
                map[i][j] = (map[i - 1][j] + map[i][j - 1]) % MOD;
            }
        }
    }
    return map[m][n];
}